### PR TITLE
[processing] Add a cluster size attribute to the dbscan cluster algorithm

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR adds a new cluster size attribute attached to output features of the DBSCAN cluster algorithm. The value represents the number of features within an identified cluster. Useful attribute in many cases.

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmdbscanclustering.cpp [hunk 1]
```diff
@@ -133,11 +139,13 @@ QVariantMap QgsDbscanClusteringAlgorithm::processAlgorithm( const QVariantMap &p
   feedback->pushInfo( QObject::tr( "Analysing clusters" ) );
   std::unordered_map< QgsFeatureId, int> idToCluster;
   idToCluster.reserve( index.size() );
+  std::unordered_map< int, int> clusterSize;
   QgsFeatureIterator features = source->getFeatures( QgsFeatureRequest().setNoAttributes() );
   const long featureCount = source->featureCount();
+  dbscan( minSize, eps, borderPointsAreNoise, featureCount, features, index, idToCluster, clusterSize, feedback );
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmdbscanclustering.cpp:145]: passing clusterSize here shouldn't be required -- dbscan doesn't do anything useful with it

Comment (nirvn) [reply src/analysis/processing/qgsalgorithmdbscanclustering.cpp:145]: I'm passing it on so I can initiate the map with all cluster IDs and a count of 0. That allows for a nice one line loop to compute the  cluster size afterwards. You prefer if I'd go at it another way?

Comment (nyalldawson) [reply src/analysis/processing/qgsalgorithmdbscanclustering.cpp:145]: It shouldn't be needed -- the call to ` clusterSize[ idCluster.second ]` would automatically add the missing id anyway and initialize it to 0.

Comment (nirvn) [reply src/analysis/processing/qgsalgorithmdbscanclustering.cpp:145]: TIL moment, thanks

**Comment by nirvn** [comment]
@nyalldawson , comments addressed.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**